packages/block-editor/src/components/inspector-controls-tabs/styles-tab.js

```diff
@@ -3,19 +3,65 @@
  */
 import { PanelBody } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import { useDispatch, useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import BlockStyles from '../block-styles';
 import InspectorControls from '../inspector-controls';
 import { useBorderPanelLabel } from '../../hooks/border';
+import { useBlockSettings } from '../../hooks/utils';
+import { store as blockEditorStore } from '../../store';
+import { ColorEdit } from '../../hooks/color';
+import { ColorToolsPanel } from '../global-styles/color-panel';
+
+function SectionBlockControls( { blockName, clientId, contentClientIds } ) {
+	const settings = useBlockSettings( blockName );
+	const { updateBlockAttributes } = useDispatch( blockEditorStore );
+
+	const { hasButton, hasHeading } = useSelect(
+		( select ) => {
+			const blockNames =
+				select( blockEditorStore ).getBlockNamesByClientId(
+					contentClientIds
+				);
+			return {
+				hasButton: blockNames.includes( 'core/button' ),
+				hasHeading: blockNames.includes( 'core/heading' ),
```

Comment:
What do you think about removing the control and the style when the element is deleted?

From what I can see, if the element is deleted inside the pattern, the style control persists when a value is set.

https://github.com/user-attachments/assets/09e8ea88-4bf8-4708-8a0e-6d9a6eabc105

I'm not sure it's a huge problem, so more of a UX question.

Reply:
I'm not sure we should be allowing deletion of elements in contentOnly!

Reply:
> I'm not sure we should be allowing deletion of elements in contentOnly!

My understanding was also that things wouldn't be able to be deleted.

>  if the element is deleted inside the pattern, the style control persists when a value is set.

Either way, if we can delete, we should be able to undo to add it back. If we remove the set value, then they undo the change, I'd want the style to still be applied. So, I think leaving the value set is fine rather than removing it in this case.